tiledb/api/c_api/rest/test/unit_capi_rest_profile.cc

```diff
@@ -0,0 +1,101 @@
+/**
+ * @file tiledb/api/c_api/rest/test/unit_capi_rest_profile.cc
+ *
+ * @section LICENSE
+ *
+ * The MIT License
+ *
+ * @copyright Copyright (c) 2025 TileDB, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ * Validates the arguments for the RestProfile C API.
+ */
+
+#define CATCH_CONFIG_MAIN
+#include <test/support/tdb_catch.h>
+
+#include "test/support/src/temporary_local_directory.h"
+
+#include "../rest_profile_api_external.h"
+#include "../rest_profile_api_internal.h"
+
+const char* name_ = tiledb::sm::RestProfile::DEFAULT_NAME.c_str();
+tiledb::sm::TemporaryLocalDirectory tempdir_("unit_capi_rest_profile");
+
+/**
+ * No TEST_CASE "C API: tiledb_rest_profile_alloc argument validation" here;
+ * The user-facing API is not to be used in-test, to avoid async changes to
+ * the local, on-disk file. Instead, `tiledb_rest_profile_alloc_test` is used
+ * throughout this test, including in ordinary_rest_profile.
+ */
+
+TEST_CASE(
+    "C API: tiledb_rest_profile_alloc_test argument validation",
+    "[capi][rest_profile]") {
+  capi_return_t rc;
+  tiledb_rest_profile_t* profile;
+  auto homedir = tempdir_.path().c_str();
+  SECTION("success") {
+    rc = tiledb_rest_profile_alloc_test(name_, homedir, &profile);
```

Comment:
I think it's better to have more verbose or repetitive tests while testing the actual APIs, rather than testing the logic of test APIs/functions. Let's actually call the real alloc and free APIs in our tests so that we are sure and clear on what we are testing.

Reply:
I think a good endorsement of this comment is the `null name` test below which IIRC is at odds with the non-test version of the API (which substitutes a default if the name is null).